raven/util.lua

```diff
@@ -0,0 +1,145 @@
+--- Utility functions module.
+-- This is mostly used for internal stuff and should not be relevant for end
+-- users (except some function that can be overridden).
+--
+-- @module raven.util
+-- @copyright 2014-2017 CloudFlare, Inc.
+-- @license BSD 3-clause (see LICENSE file)
+
+local string_format = string.format
+local string_find = string.find
+local string_sub = string.sub
+local string_match = string.match
+local math_random = math.random
+local os_date = os.date
+
+local _M = {}
+
+local _VERSION = "0.5.0"
+_M._VERSION = _VERSION
+
+--- Used to log errors during reporting.
+-- The default implementation is quite dumb: it will simply use `print` to
+-- display the error message. Users are encouraged to override this
+-- implementation to something smarter.
+-- @param ... Message to log (will be concatenated)
+function _M.errlog(...)
+   print("[ERROR]", ...)
+end
+
+--- Create a UUID in Version 4 format as a string albeit without the `-s`
+-- separating fields.
+function _M.uuid4()
+   -- TODO: the Sentry SDK docs don't specify any requirement for `event_id`
+   -- field as far as I know
+
+   -- we need 12 random hex digits at some point, but some version of Lua can
+   -- only generate random integers up to 2^31, so we are limited to 7
+   -- hex-digits per call
+   return string_format("%07x%05x4%03x8%03x%07x%05x",
```

Comment:
well like I said in the comment above, the previous version used to generate (inefficiently) a "uuid" this way, but the [Sentry docs](https://docs.sentry.io/clientdev/attributes/) don't have such a requirement in the id. I'll change the code to generate a plain random hexstring instead.